core/tee/tee_pobj.c

```diff
@@ -64,14 +65,36 @@ TEE_Result tee_pobj_get(TEE_UUID *uuid, void *obj_id, uint32_t obj_id_len,
 {
 	TEE_Result res = TEE_SUCCESS;
 	struct tee_pobj *o = NULL;
+	void *obj_id_kbuf = NULL;
+	bool obj_id_from_user = true;
 
 	*obj = NULL;
 
+	res = memdup_user(obj_id, obj_id_len, &obj_id_kbuf);
+	switch (res) {
+	case TEE_SUCCESS:
+		break;
```

Comment:
explicit `obj_id_from_user = true;` here would be better IMHO

Reply:
updated